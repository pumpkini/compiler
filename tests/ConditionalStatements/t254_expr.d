int main()  {
    if(itob(1+2)){
        Print(1);
    }else{
        Print("wrong");
    }

    if(itob(1 - 1)){
        Print("wrong");
    }else{
        Print(2);
    }

    if(itob(5*2)){
        Print(3);
    }else{
        Print("wrong");
    }

    if(itob(10*0)) {
        Print("wrong");
    }else{
        Print(4);
    }

    if (3.14 > 3.13){
        Print(5);
    }

    if (3.14 >= 3.14){
        Print(6);
    }

    if (2.5*3.0 >= 5.0){
        Print(7);
    }

    Print(-234);
    Print(2+3);
    Print(13*3);
    Print(123-525);
    Print(12345/351);
    Print(99199%2231);
    Print(2<5);
    Print(2<=3);
    Print(2<=2);
    Print(2>5);
    Print(2>=3);
    Print(2>=2);
    Print(2==3);
    Print(2!=3);
    Print(2==2);
    Print(true&&true);
    Print(false&&true);
    Print(true||true);
    Print(false||true);
    Print(false||false);
    Print(!(false));
    Print(2234*12*4/24+35-12341+52*12%123+123-521*123/12234*1+41/4+2);
    Print(41*334+169%478/962%705+281*961-995*827-391%902/292*421-718+447-771/869*667%35%703-322/673%141-253/547%662*37%723-529/316+190*288-40*264/446+890%370+6-393);
    Print(41*334+169%478/962%705+281*961-995*827-391%902/292*421-718+447-771/869*667%35%703-322/673%141-253/547%662*37%723-529/316+190*288-40*264/446+890%370+6-393/629/84%756+966-931/944%626/537/118*929-833+639/704+977-673-21+924*270%777/97*986+161-355*655%31*350+941%966+107-7*457*753/945%209/221/422-506+413/900-762+410%624*548/595-602+291-374+596-348%668%281%53%418/900/127*728/648/807-310*813);
    Print(((((42/19-39-10)/(46*79+74))/((81+8)+(18-77+47))+((82%69/12)+(29*37))+((11%6)%(63)))*(((23*53+92)+(66*48))%((80+25)/(74/78+34-32)-(57/52%64)/(93+1))*((11-15-32-67)+(39)/(1-90))%((2+93)))));
    Print(((((8*38/48)%(71*51*97)/(84/82*98))%((57-90)%(45))/((100*51*8)))));
    Print(((((48+82+1)+(84/69/51/43))+((4*25%39-5)-(85*24%26+46)))));
    Print(((((90%38/77/82)-(15/87)*(91)*(62*35))+((80)*(14%48)))));
    Print(((((100.68/33.99*70.81)-(31.69-6.52*14.78)+(22.36-52.21+80.41+86.57)+(75.12-86.24*55.46/74.48))+((53.86)))+(((48.76-30.3/23.41))*((84.31-56.89))-((67.14)/(59.57+16.15))))>1776.8);
    Print(((((100.68/33.99*70.81)-(31.69-6.52*14.78)+(22.36-52.21+80.41+86.57)+(75.12-86.24*55.46/74.48))+((53.86)))+(((48.76-30.3/23.41))*((84.31-56.89))-((67.14)/(59.57+16.15))))<1779.1);
    Print(((((13.54+20.43))-((28.66)-(95.13))-((24.56/12.86-80.35*28.3)-(13.36+92.64)-(92.87/73.66/55.45*58.72))))>2479.07);
    Print(((((13.54+20.43))-((28.66)-(95.13))-((24.56/12.86-80.35*28.3)-(13.36+92.64)-(92.87/73.66/55.45*58.72))))<2480.07);

}
