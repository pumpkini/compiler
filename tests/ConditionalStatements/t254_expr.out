1
2
3
4
5
6
7
-234
5
39
-402
35
1035
true
true
true
false
false
true
false
true
true
true
false
true
true
false
true
-7699
-485363
-745105
-60424
0
90
1120
true
true
true
true
